js/src/generate/service-gen.ts

```diff
@@ -0,0 +1,97 @@
+  );
+};
+
+const getAccount = (kind: MethodType) => {
+  return kind === 'message' ? `account: ${HEX_STRING_TYPE} | IKeyringPair` : '';
+};
+
+export class ServiceGenerator {
+  private scaleTypes: Record<string, any> = {};
+  constructor(private _out: Output) {}
+
+  public generate(service: IService, scaleTypes: Record<string, any>) {
+    this.scaleTypes = scaleTypes;
+    this._out
+      .import('@gear-js/api', 'GearApi')
+      .import('./transaction.js', 'Transaction')
+      .block(`export class ${getClassName(service.name)} extends Transaction`, () => {
+        this._out.block(`constructor(api: GearApi, public programId: ${HEX_STRING_TYPE})`, () => {
+          this._out
+            .block(`const types: Record<string, any> =`, () => {
+              for (const [name, type] of Object.entries(this.scaleTypes)) {
+                this._out.line(`${name}: ${JSON.stringify(type)},`, false);
+              }
+            })
+            .line('super(api, types)');
+        });
+        this.generateMethods(service.methods);
+      });
+  }
+
+  private generateMethods(methods: IServiceMethod[]) {
+    this._out.import('@polkadot/types/types', 'IKeyringPair');
+
+    let messageIndex = 0;
+    let queryIndex = 0;
+    for (const {
+      def: { args, name, output },
+      kind,
+    } of methods) {
+      const returnType = getTypeDef(output, false, true);
+
+      this._out
+        .line()
+        .block(
+          `public async ${name[0].toLowerCase() + name.slice(1)}(${getArgs(args)}${getAccount(
+            kind,
+          )}): Promise<${returnType}>`,
+          () => {
+            let index = kind === 'message' ? messageIndex : queryIndex;
+            if (args.length === 0) {
+              this._out.line(`const payload = this.registry.createType('u8', ${index}).toU8a()`);
+            } else {
+              this._out
+                .line(`const payload = [`, false)
+                .increaseIndent()
+                .line(`...this.registry.createType('u8', ${index}).toU8a(),`, false);
+              for (const { name, type } of args) {
+                this._out.line(`...this.registry.createType('${getType(type, true)}', ${name}).toU8a(),`, false);
+              }
+              this._out.reduceIndent().line(']');
+            }
+
+            if (kind === 'message') {
+              messageIndex++;
+              this._out
+                .line(`const replyPayloadBytes = await this.submitMsgAndWaitForReply(`, false)
+                .increaseIndent()
+                .line('this.programId,', false)
+                .line('payload,', false)
+                .line('account,', false)
+                .reduceIndent()
+                .line(')')
+                .line(`const result = this.registry.createType('${getType(output, true)}', replyPayloadBytes)`)
+                .line(`return result.toJSON() as ${returnType}`);
```

Comment:
`.toJSON`, as we've encountered in the past, is parsing some numeric types to hex. Wouldn't this be a problem?

Reply:
Refactored. Will return number or bigint in case of numeric types and string in case of string type